src/Assets/LightSkyDefense/Scripts/SpawnCreditOnDestruction.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class SpawnCreditOnDestruction : MonoBehaviour
+{
+    public Credit Credit;
+    public int Value;
+
+    private bool isQuitting = false;
+    void OnApplicationQuit()
+    {
+        isQuitting = true;
+    }
+
+    /// <summary>
+    /// Spawn money when its gameObject is destroyed.
+    /// </summary>
+    void OnDisable()
+    {
+        Credit.Value = Value;
```

Comment:
We create two new Credits, do we do this on purpose?

Reply:
no